### DISCUSSION TITLE
Read Status

### DISCUSSION THREAD

**Discussion by textbrowser** [textbrowser/biblioteq#85] [Ideas]
ksl1989

Thank you for your work. I've been looking for some time and I've not seen any comparable application in terms of features/ease of use.

However, one of the main features I need is missing: I'd like to be able to set a book as read/unread. One of the reasons I'm looking to organize my books is that I keep forgetting books I haven't read, so I'd be able to list/filter unready books.

I understand it's possible to achieve a possible result by adding myself as a member and checking out books to me, but that feels unnecessarily complicated when I'm using this application to organize my personal collection. I'd just like to be able to set a book as read or unread when I add it and be able to update it to read in the book entry dialogue.

**Comment by ksl1989** [comment]
In reply to your reply: I, of course, agree a "read staus" feature isn't relevant to libraries. But since I imagine many of the potential users of this application are people just trying to organize their personal book collection, you might want to add a "private" or "personal" mode, with options that would be useful for these users. I suspect more people would find it useful — for me, it's one of the main reasons I've been looking into collection managers.

But adding categories would work for me, as long as I'd be able to list books in one category. Especially if you are thinking of other uses for categories, this may be a very useful feature.

In any case, thanks for considering it.

**Comment by textbrowser** [comment]
For PQ patrons: a new private field. For SQLite, a new field. The field indicates if the CD or book or whatever was processed by the person. A boolean is insufficient, so it has to be a text field. Since it's a text field, it has to be configurable. Since it's configurable, it has to be configurable by the user. New enumeration. Yup, lots of work. Scheduled for next year. Just added a new field and that cascaded into an avalanche of changes.

How about a nice pencil and paper? :)

**Comment by textbrowser** [comment]
Alive? Where would you like the read status field?

- add an icon to a read item in Icons Mode view
- new All Read category
- new column in Table Mode view
- new status text in Summary panel (left-hand side)
- new highlight color denoting read status
- something else

**Comment by ksl1989** [comment]
@textbrowser Sorry, I'm alive for the most part, just haven't had the time to do or read much of anything. Thanks for thinking and working on this. I don't really have a strong opinion about where the field should show, as long as I can look it up and filter books by this property. I think it's reasonable to show as column and in the summary panel, as most similar fields show there (although ideally the fields shown should be configurable). I'd be more interested in an All Unread than an All Read category, as these are the books I want to not overlook. I wouldn't want a highlight colour, but I wouldn't mind it as long as it's optional.

**Comment by textbrowser** [comment]
I'm sorry for the delay. Hope life is better. I am introducing a book_read database column. The new feature will be available with the 2021.08.08 release.

**Reply by ksl1989** [reply]
Thank you for this @textbrowser! I intend to use it once this version comes out.

**Comment by textbrowser** [comment]
It will be available only for SQLite. The Title column is now check-enabled and the status is included in the summary.

**Reply by ksl1989** [reply]
Yes, I have been using SQLite, I'll definitely try it when you release it.

**Comment by textbrowser** [comment]
For icon-view, I will have to implement a new approach... a sort of check-enabled image. Not sure if this will be available in the next release.

**Comment by textbrowser** [comment]
Did you test the new read-status?

**Reply by ksl1989** [reply]
I did test it. It works for me, does all that I need. But I'm not sure if this is the best design choice; generally, a checkbox to the left of a list means select, and that's what most users will think it is, a select box. Even though I know what it is for, I still sometimes look at it and think "huh, why are those books selected?". Just something to keep in mind.

For my part, it does what I need it to do, and I can only thank you to adding this feature. When I have some time, I'll try to add more books and I may then have more issues/suggestions — but that may take some time.

**Comment by textbrowser** [comment]
How about a separate column?

**Reply by ksl1989** [reply]
Yes, I think a separate column would be less confusing for most people.

**Comment by textbrowser** [comment]
New column added.

**Comment by textbrowser** [comment]
Uploading the new releases today. :)

**Comment by textbrowser** [comment]
Was a challenging change because the column does not exist in PostgreSQL and was afraid of damaging existing queries.

**Comment by textbrowser** [comment]
![Screenshot_20210910_101744](https://user-images.githubusercontent.com/10701156/132867960-010f9e23-3937-49e8-896b-8d08dbe3ad0f.png)

**Reply by ksl1989** [reply]
Looks good! I feel it's much clearer this way.

**Comment by textbrowser** [comment]
It was a simple request with a complicated solution. :P

**Comment by CeruleanPhoenix** [comment]
Hello,

I'm running version 2021.10.30. Is it possible to use as a search parameter/filter the new "Read Status" feature from within the app. If not, is this a planned feature? Also, I think that for uniformity reasons an option to modify "Read status" should be visible from within the "Modify Book Entry" panel.

Thank you for your hard work.

**Comment by textbrowser** [comment]
No, not yet. I'll see what I can do. :)
